### ISSUE TITLE
Nightly fastlane betas

### ISSUE THREAD

**Issue by KrauseFx** [fastlane/fastlane#8121]
To avoid introducing regressions we would like to introduce a nightly fastlane beta channel. You add one line to your `Fastfile`, and automatically get the latest build from master, built every night for you.

This way we can detect regressions earlier, before rolling out changes to 100% of the user.

**The question**: Would you use it and help report if a new issue got introduced? In general the code on `master` is stable and was reviewed.

**Comment by lyndsey-ferguson** [comment]
Was the 'colored' issue due to an incompatibility with a different version of Ruby? Any way to have tests running against the different versions of Ruby or is that too large a target that would make this impossible?

**Comment by KrauseFx** [comment]
@lyndsey-ferguson no, this was an issue that only occurred on certain parameters, nothing to do with the Ruby version. With Circle we can't run matrix projects, so we can't really test multiple Ruby versions in parallel

**Comment by lyndsey-ferguson** [comment]
Okay. Should we have tests around those parameters then? Could it be more reliable than a beta program..

**Comment by hjanuschka** [comment]
i am a huge fan of `-pre`  releases based on latest master, i would go one step further and do a `-pre` on every master commit.

**Comment by PetrosIchor** [comment]
Yeah, I'd totally use it.  At the moment, I'm pulling weekly as I put together 2 new build systems. Having a more rapid updates would add pressure to keep an eye on weak/unfinished components. Plus I get to help create a better feedback loop with y'all. 

However, I do agree with hjanuschka's concern about velocity. It could put an unintentional barrier between relase versions. I remember awhile back y'all had metrics tracking that could be disabled with a flag?  Could adding metrics tracking to new work help fill that gap?

If the intention is to get more data on regressions and community pain points, than make sure the change in process provides enough value. Having some anonymous usage data might help fill the gaps of mirco-release development.  If development does begin to bottleneck, at least the quality of output and the effectiveness of new features will be improved.

Or maybe that's a huge lift and I'm talking out my ass. lol. 

Oh, kickass on making enterprise a first class citizen in match! It's great getting to remove a step in a flow and have no impact. Keep up the great work!

**Comment by mgrebenets** [comment]
Sounds like a good idea to me. I could setup one build on our TeamCity to run at midnight and pull latest fastlane beta.

This will not cover all of the fastlane functionality, of course, we only use things like `scan`, `gym`, few other actions and plugins. 

Actually, I could also setup a nightly resigner CI job, so it would cover some of the `sigh resign` part.

It would be helpful, things like recent shadowing of short args for `sigh resign` would be caught earlier.

I have to double-check if our TeamCity would be able to pull from GitHub though. We are getting Ruby gems from Artifactory's rubygems.org mirror.

But if beta or `pre-` is published on rubygems.org, then there will be no issue I can think of.

Could it be similar to what CocoaPods does?
They show the message about available prerelease and ask to help testing it.

I'm not that good with ruby and bundler, but I believe there's some in-built support for `pre-` prefixes while setting up gems.

**Comment by KrauseFx** [comment]
This is now implemented with [krausefx/nigthly](https://github.com/krausefx/nightly), see the published releases on [rubygems.org/gems/fastlane](https://rubygems.org/gems/fastlane) 🚀

We'll still need an easy way for people to switch to nightly builds, but for now, people can manually update to those releases 👍
